packages/orm/src/modelDescriptorsRegistry/modelDescriptorsRegistry.ts

```diff
@@ -0,0 +1,50 @@
+import { attr, Attribute } from "../fields";
+import { Descriptors } from '../types';
+
+/**
+ * Represents a map with descriptors for a specific model
+ */
+type DescriptorsMap<DescriptorTypes extends Descriptors> = { id: Attribute } & { [DescriptorName: string]: DescriptorTypes }
+/**
+ * Stores descriptors maps for each registered model
+ */
+type Registry<DescriptorTypes extends Descriptors = Descriptors> = { [ModelName: string]: DescriptorsMap<DescriptorTypes> };
+
+/**
+ * A singleton storing descriptors for each registered model.
+ */
+export class ModelDescriptorsRegistry {
```

Comment:
I lifted the type here as I find it redundant to couple it with the some generic schema type that could be passed explicitly at every instance usage.